// @aws-sdk/client-route-53/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateHealthCheckRequest = exports.HealthCheckConfig = exports.HealthCheckType = exports.CreateCidrCollectionResponse = exports.CidrCollection = exports.CreateCidrCollectionRequest = exports.CidrCollectionAlreadyExistsException = exports.ThrottlingException = exports.ChangeTagsForResourceResponse = exports.ChangeTagsForResourceRequest = exports.Tag = exports.NoSuchHealthCheck = exports.InvalidChangeBatch = exports.ChangeResourceRecordSetsResponse = exports.ChangeResourceRecordSetsRequest = exports.ChangeBatch = exports.Change = exports.ResourceRecordSet = exports.ResourceRecord = exports.GeoLocation = exports.CidrRoutingConfig = exports.NoSuchCidrCollectionException = exports.CidrCollectionVersionMismatchException = exports.CidrBlockInUseException = exports.ChangeCidrCollectionResponse = exports.ChangeCidrCollectionRequest = exports.CidrCollectionChange = exports.CidrCollectionChangeAction = exports.PublicZoneVPCAssociation = exports.PriorRequestNotComplete = exports.NotAuthorizedException = exports.NoSuchHostedZone = exports.LimitsExceeded = exports.InvalidVPCId = exports.ConflictingDomainExists = exports.AssociateVPCWithHostedZoneResponse = exports.AssociateVPCWithHostedZoneRequest = exports.VPC = exports.AliasTarget = exports.AlarmIdentifier = exports.NoSuchKeySigningKey = exports.InvalidSigningStatus = exports.InvalidKMSArn = exports.InvalidKeySigningKeyStatus = exports.InvalidInput = exports.ConcurrentModification = exports.ActivateKeySigningKeyResponse = exports.ChangeInfo = exports.ActivateKeySigningKeyRequest = exports.AccountLimit = void 0;
exports.CreateTrafficPolicyVersionResponse = exports.CreateTrafficPolicyVersionRequest = exports.TrafficPolicyInstanceAlreadyExists = exports.TooManyTrafficPolicyInstances = exports.NoSuchTrafficPolicy = exports.CreateTrafficPolicyInstanceResponse = exports.TrafficPolicyInstance = exports.CreateTrafficPolicyInstanceRequest = exports.TrafficPolicyAlreadyExists = exports.TooManyTrafficPolicies = exports.InvalidTrafficPolicyDocument = exports.CreateTrafficPolicyResponse = exports.TrafficPolicy = exports.CreateTrafficPolicyRequest = exports.HostedZoneNotFound = exports.DelegationSetAlreadyReusable = exports.DelegationSetAlreadyCreated = exports.CreateReusableDelegationSetResponse = exports.CreateReusableDelegationSetRequest = exports.QueryLoggingConfigAlreadyExists = exports.NoSuchCloudWatchLogsLogGroup = exports.InsufficientCloudWatchLogsResourcePolicy = exports.CreateQueryLoggingConfigResponse = exports.QueryLoggingConfig = exports.CreateQueryLoggingConfigRequest = exports.TooManyKeySigningKeys = exports.KeySigningKeyAlreadyExists = exports.InvalidKeySigningKeyName = exports.InvalidArgument = exports.CreateKeySigningKeyResponse = exports.KeySigningKey = exports.CreateKeySigningKeyRequest = exports.TooManyHostedZones = exports.NoSuchDelegationSet = exports.InvalidDomainName = exports.HostedZoneAlreadyExists = exports.DelegationSetNotReusable = exports.DelegationSetNotAvailable = exports.CreateHostedZoneResponse = exports.HostedZone = exports.DelegationSet = exports.CreateHostedZoneRequest = exports.HostedZoneConfig = exports.TooManyHealthChecks = exports.HealthCheckAlreadyExists = exports.CreateHealthCheckResponse = exports.HealthCheck = exports.LinkedService = exports.CloudWatchAlarmConfiguration = exports.Dimension = void 0;
exports.NoSuchChange = exports.GetChangeResponse = exports.GetChangeRequest = exports.GetAccountLimitResponse = exports.GetAccountLimitRequest = exports.KeySigningKeyWithActiveStatusNotFound = exports.HostedZonePartiallyDelegated = exports.EnableHostedZoneDNSSECResponse = exports.EnableHostedZoneDNSSECRequest = exports.VPCAssociationNotFound = exports.LastVPCAssociation = exports.DisassociateVPCFromHostedZoneResponse = exports.DisassociateVPCFromHostedZoneRequest = exports.DNSSECNotFound = exports.DisableHostedZoneDNSSECResponse = exports.DisableHostedZoneDNSSECRequest = exports.VPCAssociationAuthorizationNotFound = exports.DeleteVPCAssociationAuthorizationResponse = exports.DeleteVPCAssociationAuthorizationRequest = exports.NoSuchTrafficPolicyInstance = exports.DeleteTrafficPolicyInstanceResponse = exports.DeleteTrafficPolicyInstanceRequest = exports.TrafficPolicyInUse = exports.DeleteTrafficPolicyResponse = exports.DeleteTrafficPolicyRequest = exports.DeleteReusableDelegationSetResponse = exports.DeleteReusableDelegationSetRequest = exports.DelegationSetInUse = exports.NoSuchQueryLoggingConfig = exports.DeleteQueryLoggingConfigResponse = exports.DeleteQueryLoggingConfigRequest = exports.DeleteKeySigningKeyResponse = exports.DeleteKeySigningKeyRequest = exports.HostedZoneNotEmpty = exports.DeleteHostedZoneResponse = exports.DeleteHostedZoneRequest = exports.HealthCheckInUse = exports.DeleteHealthCheckResponse = exports.DeleteHealthCheckRequest = exports.DeleteCidrCollectionResponse = exports.DeleteCidrCollectionRequest = exports.CidrCollectionInUseException = exports.KeySigningKeyInUse = exports.KeySigningKeyInParentDSRecord = exports.DeactivateKeySigningKeyResponse = exports.DeactivateKeySigningKeyRequest = exports.TooManyVPCAssociationAuthorizations = exports.CreateVPCAssociationAuthorizationResponse = exports.CreateVPCAssociationAuthorizationRequest = exports.TooManyTrafficPolicyVersionsForCurrentPolicy = void 0;
exports.LocationSummary = exports.ListCidrLocationsRequest = exports.ListCidrCollectionsResponse = exports.CollectionSummary = exports.ListCidrCollectionsRequest = exports.NoSuchCidrLocationException = exports.ListCidrBlocksResponse = exports.CidrBlockSummary = exports.ListCidrBlocksRequest = exports.GetTrafficPolicyInstanceCountResponse = exports.GetTrafficPolicyInstanceCountRequest = exports.GetTrafficPolicyInstanceResponse = exports.GetTrafficPolicyInstanceRequest = exports.GetTrafficPolicyResponse = exports.GetTrafficPolicyRequest = exports.GetReusableDelegationSetLimitResponse = exports.ReusableDelegationSetLimit = exports.GetReusableDelegationSetLimitRequest = exports.GetReusableDelegationSetResponse = exports.GetReusableDelegationSetRequest = exports.GetQueryLoggingConfigResponse = exports.GetQueryLoggingConfigRequest = exports.HostedZoneNotPrivate = exports.GetHostedZoneLimitResponse = exports.HostedZoneLimit = exports.GetHostedZoneLimitRequest = exports.GetHostedZoneCountResponse = exports.GetHostedZoneCountRequest = exports.GetHostedZoneResponse = exports.GetHostedZoneRequest = exports.GetHealthCheckStatusResponse = exports.GetHealthCheckStatusRequest = exports.GetHealthCheckLastFailureReasonResponse = exports.HealthCheckObservation = exports.StatusReport = exports.GetHealthCheckLastFailureReasonRequest = exports.GetHealthCheckCountResponse = exports.GetHealthCheckCountRequest = exports.IncompatibleVersion = exports.GetHealthCheckResponse = exports.GetHealthCheckRequest = exports.NoSuchGeoLocation = exports.GetGeoLocationResponse = exports.GeoLocationDetails = exports.GetGeoLocationRequest = exports.GetDNSSECResponse = exports.DNSSECStatus = exports.GetDNSSECRequest = exports.GetCheckerIpRangesResponse = exports.GetCheckerIpRangesRequest = void 0;
exports.UpdateTrafficPolicyInstanceResponse = exports.UpdateTrafficPolicyInstanceRequest = exports.ConflictingTypes = exports.UpdateTrafficPolicyCommentResponse = exports.UpdateTrafficPolicyCommentRequest = exports.UpdateHostedZoneCommentResponse = exports.UpdateHostedZoneCommentRequest = exports.UpdateHealthCheckResponse = exports.UpdateHealthCheckRequest = exports.HealthCheckVersionMismatch = exports.TestDNSAnswerResponse = exports.TestDNSAnswerRequest = exports.ListVPCAssociationAuthorizationsResponse = exports.ListVPCAssociationAuthorizationsRequest = exports.ListTrafficPolicyVersionsResponse = exports.ListTrafficPolicyVersionsRequest = exports.ListTrafficPolicyInstancesByPolicyResponse = exports.ListTrafficPolicyInstancesByPolicyRequest = exports.ListTrafficPolicyInstancesByHostedZoneResponse = exports.ListTrafficPolicyInstancesByHostedZoneRequest = exports.ListTrafficPolicyInstancesResponse = exports.ListTrafficPolicyInstancesRequest = exports.ListTrafficPoliciesResponse = exports.TrafficPolicySummary = exports.ListTrafficPoliciesRequest = exports.ListTagsForResourcesResponse = exports.ListTagsForResourcesRequest = exports.ListTagsForResourceResponse = exports.ResourceTagSet = exports.ListTagsForResourceRequest = exports.ListReusableDelegationSetsResponse = exports.ListReusableDelegationSetsRequest = exports.ListResourceRecordSetsResponse = exports.ListResourceRecordSetsRequest = exports.ListQueryLoggingConfigsResponse = exports.ListQueryLoggingConfigsRequest = exports.ListHostedZonesByVPCResponse = exports.HostedZoneSummary = exports.HostedZoneOwner = exports.ListHostedZonesByVPCRequest = exports.InvalidPaginationToken = exports.ListHostedZonesByNameResponse = exports.ListHostedZonesByNameRequest = exports.ListHostedZonesResponse = exports.ListHostedZonesRequest = exports.ListHealthChecksResponse = exports.ListHealthChecksRequest = exports.ListGeoLocationsResponse = exports.ListGeoLocationsRequest = exports.ListCidrLocationsResponse = void 0;
const Route53ServiceException_1 = require("./Route53ServiceException");
var AccountLimit;
(function (AccountLimit) {
    AccountLimit.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccountLimit = exports.AccountLimit || (exports.AccountLimit = {}));
var ActivateKeySigningKeyRequest;
(function (ActivateKeySigningKeyRequest) {
    ActivateKeySigningKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ActivateKeySigningKeyRequest = exports.ActivateKeySigningKeyRequest || (exports.ActivateKeySigningKeyRequest = {}));
var ChangeInfo;
(function (ChangeInfo) {
    ChangeInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeInfo = exports.ChangeInfo || (exports.ChangeInfo = {}));
var ActivateKeySigningKeyResponse;
(function (ActivateKeySigningKeyResponse) {
    ActivateKeySigningKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ActivateKeySigningKeyResponse = exports.ActivateKeySigningKeyResponse || (exports.ActivateKeySigningKeyResponse = {}));
class ConcurrentModification extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "ConcurrentModification",
            $fault: "client",
            ...opts,
        });
        this.name = "ConcurrentModification";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConcurrentModification.prototype);
    }
}
exports.ConcurrentModification = ConcurrentModification;
class InvalidInput extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidInput",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidInput";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidInput.prototype);
    }
}
exports.InvalidInput = InvalidInput;
class InvalidKeySigningKeyStatus extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidKeySigningKeyStatus",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidKeySigningKeyStatus";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidKeySigningKeyStatus.prototype);
    }
}
exports.InvalidKeySigningKeyStatus = InvalidKeySigningKeyStatus;
class InvalidKMSArn extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidKMSArn",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidKMSArn";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidKMSArn.prototype);
    }
}
exports.InvalidKMSArn = InvalidKMSArn;
class InvalidSigningStatus extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidSigningStatus",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidSigningStatus";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidSigningStatus.prototype);
    }
}
exports.InvalidSigningStatus = InvalidSigningStatus;
class NoSuchKeySigningKey extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchKeySigningKey",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchKeySigningKey";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchKeySigningKey.prototype);
    }
}
exports.NoSuchKeySigningKey = NoSuchKeySigningKey;
var AlarmIdentifier;
(function (AlarmIdentifier) {
    AlarmIdentifier.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AlarmIdentifier = exports.AlarmIdentifier || (exports.AlarmIdentifier = {}));
var AliasTarget;
(function (AliasTarget) {
    AliasTarget.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AliasTarget = exports.AliasTarget || (exports.AliasTarget = {}));
var VPC;
(function (VPC) {
    VPC.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VPC = exports.VPC || (exports.VPC = {}));
var AssociateVPCWithHostedZoneRequest;
(function (AssociateVPCWithHostedZoneRequest) {
    AssociateVPCWithHostedZoneRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AssociateVPCWithHostedZoneRequest = exports.AssociateVPCWithHostedZoneRequest || (exports.AssociateVPCWithHostedZoneRequest = {}));
var AssociateVPCWithHostedZoneResponse;
(function (AssociateVPCWithHostedZoneResponse) {
    AssociateVPCWithHostedZoneResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AssociateVPCWithHostedZoneResponse = exports.AssociateVPCWithHostedZoneResponse || (exports.AssociateVPCWithHostedZoneResponse = {}));
class ConflictingDomainExists extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "ConflictingDomainExists",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictingDomainExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictingDomainExists.prototype);
    }
}
exports.ConflictingDomainExists = ConflictingDomainExists;
class InvalidVPCId extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidVPCId",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidVPCId";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidVPCId.prototype);
    }
}
exports.InvalidVPCId = InvalidVPCId;
class LimitsExceeded extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "LimitsExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "LimitsExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, LimitsExceeded.prototype);
    }
}
exports.LimitsExceeded = LimitsExceeded;
class NoSuchHostedZone extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchHostedZone",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchHostedZone";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchHostedZone.prototype);
    }
}
exports.NoSuchHostedZone = NoSuchHostedZone;
class NotAuthorizedException extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NotAuthorizedException",
            $fault: "client",
            ...opts,
        });
        this.name = "NotAuthorizedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, NotAuthorizedException.prototype);
    }
}
exports.NotAuthorizedException = NotAuthorizedException;
class PriorRequestNotComplete extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "PriorRequestNotComplete",
            $fault: "client",
            ...opts,
        });
        this.name = "PriorRequestNotComplete";
        this.$fault = "client";
        Object.setPrototypeOf(this, PriorRequestNotComplete.prototype);
    }
}
exports.PriorRequestNotComplete = PriorRequestNotComplete;
class PublicZoneVPCAssociation extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "PublicZoneVPCAssociation",
            $fault: "client",
            ...opts,
        });
        this.name = "PublicZoneVPCAssociation";
        this.$fault = "client";
        Object.setPrototypeOf(this, PublicZoneVPCAssociation.prototype);
    }
}
exports.PublicZoneVPCAssociation = PublicZoneVPCAssociation;
var CidrCollectionChangeAction;
(function (CidrCollectionChangeAction) {
    CidrCollectionChangeAction["DELETE_IF_EXISTS"] = "DELETE_IF_EXISTS";
    CidrCollectionChangeAction["PUT"] = "PUT";
})(CidrCollectionChangeAction = exports.CidrCollectionChangeAction || (exports.CidrCollectionChangeAction = {}));
var CidrCollectionChange;
(function (CidrCollectionChange) {
    CidrCollectionChange.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CidrCollectionChange = exports.CidrCollectionChange || (exports.CidrCollectionChange = {}));
var ChangeCidrCollectionRequest;
(function (ChangeCidrCollectionRequest) {
    ChangeCidrCollectionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeCidrCollectionRequest = exports.ChangeCidrCollectionRequest || (exports.ChangeCidrCollectionRequest = {}));
var ChangeCidrCollectionResponse;
(function (ChangeCidrCollectionResponse) {
    ChangeCidrCollectionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeCidrCollectionResponse = exports.ChangeCidrCollectionResponse || (exports.ChangeCidrCollectionResponse = {}));
class CidrBlockInUseException extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "CidrBlockInUseException",
            $fault: "client",
            ...opts,
        });
        this.name = "CidrBlockInUseException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CidrBlockInUseException.prototype);
        this.Message = opts.Message;
    }
}
exports.CidrBlockInUseException = CidrBlockInUseException;
class CidrCollectionVersionMismatchException extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "CidrCollectionVersionMismatchException",
            $fault: "client",
            ...opts,
        });
        this.name = "CidrCollectionVersionMismatchException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CidrCollectionVersionMismatchException.prototype);
        this.Message = opts.Message;
    }
}
exports.CidrCollectionVersionMismatchException = CidrCollectionVersionMismatchException;
class NoSuchCidrCollectionException extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchCidrCollectionException",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchCidrCollectionException";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchCidrCollectionException.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchCidrCollectionException = NoSuchCidrCollectionException;
var CidrRoutingConfig;
(function (CidrRoutingConfig) {
    CidrRoutingConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CidrRoutingConfig = exports.CidrRoutingConfig || (exports.CidrRoutingConfig = {}));
var GeoLocation;
(function (GeoLocation) {
    GeoLocation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GeoLocation = exports.GeoLocation || (exports.GeoLocation = {}));
var ResourceRecord;
(function (ResourceRecord) {
    ResourceRecord.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceRecord = exports.ResourceRecord || (exports.ResourceRecord = {}));
var ResourceRecordSet;
(function (ResourceRecordSet) {
    ResourceRecordSet.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceRecordSet = exports.ResourceRecordSet || (exports.ResourceRecordSet = {}));
var Change;
(function (Change) {
    Change.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Change = exports.Change || (exports.Change = {}));
var ChangeBatch;
(function (ChangeBatch) {
    ChangeBatch.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeBatch = exports.ChangeBatch || (exports.ChangeBatch = {}));
var ChangeResourceRecordSetsRequest;
(function (ChangeResourceRecordSetsRequest) {
    ChangeResourceRecordSetsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeResourceRecordSetsRequest = exports.ChangeResourceRecordSetsRequest || (exports.ChangeResourceRecordSetsRequest = {}));
var ChangeResourceRecordSetsResponse;
(function (ChangeResourceRecordSetsResponse) {
    ChangeResourceRecordSetsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeResourceRecordSetsResponse = exports.ChangeResourceRecordSetsResponse || (exports.ChangeResourceRecordSetsResponse = {}));
class InvalidChangeBatch extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidChangeBatch",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidChangeBatch";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidChangeBatch.prototype);
        this.messages = opts.messages;
    }
}
exports.InvalidChangeBatch = InvalidChangeBatch;
class NoSuchHealthCheck extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchHealthCheck",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchHealthCheck";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchHealthCheck.prototype);
    }
}
exports.NoSuchHealthCheck = NoSuchHealthCheck;
var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Tag = exports.Tag || (exports.Tag = {}));
var ChangeTagsForResourceRequest;
(function (ChangeTagsForResourceRequest) {
    ChangeTagsForResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeTagsForResourceRequest = exports.ChangeTagsForResourceRequest || (exports.ChangeTagsForResourceRequest = {}));
var ChangeTagsForResourceResponse;
(function (ChangeTagsForResourceResponse) {
    ChangeTagsForResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeTagsForResourceResponse = exports.ChangeTagsForResourceResponse || (exports.ChangeTagsForResourceResponse = {}));
class ThrottlingException extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
exports.ThrottlingException = ThrottlingException;
class CidrCollectionAlreadyExistsException extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "CidrCollectionAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        this.name = "CidrCollectionAlreadyExistsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CidrCollectionAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
exports.CidrCollectionAlreadyExistsException = CidrCollectionAlreadyExistsException;
var CreateCidrCollectionRequest;
(function (CreateCidrCollectionRequest) {
    CreateCidrCollectionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCidrCollectionRequest = exports.CreateCidrCollectionRequest || (exports.CreateCidrCollectionRequest = {}));
var CidrCollection;
(function (CidrCollection) {
    CidrCollection.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CidrCollection = exports.CidrCollection || (exports.CidrCollection = {}));
var CreateCidrCollectionResponse;
(function (CreateCidrCollectionResponse) {
    CreateCidrCollectionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCidrCollectionResponse = exports.CreateCidrCollectionResponse || (exports.CreateCidrCollectionResponse = {}));
var HealthCheckType;
(function (HealthCheckType) {
    HealthCheckType["CALCULATED"] = "CALCULATED";
    HealthCheckType["CLOUDWATCH_METRIC"] = "CLOUDWATCH_METRIC";
    HealthCheckType["HTTP"] = "HTTP";
    HealthCheckType["HTTPS"] = "HTTPS";
    HealthCheckType["HTTPS_STR_MATCH"] = "HTTPS_STR_MATCH";
    HealthCheckType["HTTP_STR_MATCH"] = "HTTP_STR_MATCH";
    HealthCheckType["RECOVERY_CONTROL"] = "RECOVERY_CONTROL";
    HealthCheckType["TCP"] = "TCP";
})(HealthCheckType = exports.HealthCheckType || (exports.HealthCheckType = {}));
var HealthCheckConfig;
(function (HealthCheckConfig) {
    HealthCheckConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HealthCheckConfig = exports.HealthCheckConfig || (exports.HealthCheckConfig = {}));
var CreateHealthCheckRequest;
(function (CreateHealthCheckRequest) {
    CreateHealthCheckRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateHealthCheckRequest = exports.CreateHealthCheckRequest || (exports.CreateHealthCheckRequest = {}));
var Dimension;
(function (Dimension) {
    Dimension.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Dimension = exports.Dimension || (exports.Dimension = {}));
var CloudWatchAlarmConfiguration;
(function (CloudWatchAlarmConfiguration) {
    CloudWatchAlarmConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CloudWatchAlarmConfiguration = exports.CloudWatchAlarmConfiguration || (exports.CloudWatchAlarmConfiguration = {}));
var LinkedService;
(function (LinkedService) {
    LinkedService.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LinkedService = exports.LinkedService || (exports.LinkedService = {}));
var HealthCheck;
(function (HealthCheck) {
    HealthCheck.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HealthCheck = exports.HealthCheck || (exports.HealthCheck = {}));
var CreateHealthCheckResponse;
(function (CreateHealthCheckResponse) {
    CreateHealthCheckResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateHealthCheckResponse = exports.CreateHealthCheckResponse || (exports.CreateHealthCheckResponse = {}));
class HealthCheckAlreadyExists extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "HealthCheckAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "HealthCheckAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, HealthCheckAlreadyExists.prototype);
    }
}
exports.HealthCheckAlreadyExists = HealthCheckAlreadyExists;
class TooManyHealthChecks extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TooManyHealthChecks",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyHealthChecks";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyHealthChecks.prototype);
    }
}
exports.TooManyHealthChecks = TooManyHealthChecks;
var HostedZoneConfig;
(function (HostedZoneConfig) {
    HostedZoneConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HostedZoneConfig = exports.HostedZoneConfig || (exports.HostedZoneConfig = {}));
var CreateHostedZoneRequest;
(function (CreateHostedZoneRequest) {
    CreateHostedZoneRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateHostedZoneRequest = exports.CreateHostedZoneRequest || (exports.CreateHostedZoneRequest = {}));
var DelegationSet;
(function (DelegationSet) {
    DelegationSet.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DelegationSet = exports.DelegationSet || (exports.DelegationSet = {}));
var HostedZone;
(function (HostedZone) {
    HostedZone.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HostedZone = exports.HostedZone || (exports.HostedZone = {}));
var CreateHostedZoneResponse;
(function (CreateHostedZoneResponse) {
    CreateHostedZoneResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateHostedZoneResponse = exports.CreateHostedZoneResponse || (exports.CreateHostedZoneResponse = {}));
class DelegationSetNotAvailable extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "DelegationSetNotAvailable",
            $fault: "client",
            ...opts,
        });
        this.name = "DelegationSetNotAvailable";
        this.$fault = "client";
        Object.setPrototypeOf(this, DelegationSetNotAvailable.prototype);
    }
}
exports.DelegationSetNotAvailable = DelegationSetNotAvailable;
class DelegationSetNotReusable extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "DelegationSetNotReusable",
            $fault: "client",
            ...opts,
        });
        this.name = "DelegationSetNotReusable";
        this.$fault = "client";
        Object.setPrototypeOf(this, DelegationSetNotReusable.prototype);
    }
}
exports.DelegationSetNotReusable = DelegationSetNotReusable;
class HostedZoneAlreadyExists extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "HostedZoneAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "HostedZoneAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, HostedZoneAlreadyExists.prototype);
    }
}
exports.HostedZoneAlreadyExists = HostedZoneAlreadyExists;
class InvalidDomainName extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidDomainName",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidDomainName";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidDomainName.prototype);
    }
}
exports.InvalidDomainName = InvalidDomainName;
class NoSuchDelegationSet extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchDelegationSet",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchDelegationSet";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchDelegationSet.prototype);
    }
}
exports.NoSuchDelegationSet = NoSuchDelegationSet;
class TooManyHostedZones extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TooManyHostedZones",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyHostedZones";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyHostedZones.prototype);
    }
}
exports.TooManyHostedZones = TooManyHostedZones;
var CreateKeySigningKeyRequest;
(function (CreateKeySigningKeyRequest) {
    CreateKeySigningKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateKeySigningKeyRequest = exports.CreateKeySigningKeyRequest || (exports.CreateKeySigningKeyRequest = {}));
var KeySigningKey;
(function (KeySigningKey) {
    KeySigningKey.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KeySigningKey = exports.KeySigningKey || (exports.KeySigningKey = {}));
var CreateKeySigningKeyResponse;
(function (CreateKeySigningKeyResponse) {
    CreateKeySigningKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateKeySigningKeyResponse = exports.CreateKeySigningKeyResponse || (exports.CreateKeySigningKeyResponse = {}));
class InvalidArgument extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidArgument",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidArgument";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidArgument.prototype);
    }
}
exports.InvalidArgument = InvalidArgument;
class InvalidKeySigningKeyName extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidKeySigningKeyName",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidKeySigningKeyName";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidKeySigningKeyName.prototype);
    }
}
exports.InvalidKeySigningKeyName = InvalidKeySigningKeyName;
class KeySigningKeyAlreadyExists extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "KeySigningKeyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "KeySigningKeyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, KeySigningKeyAlreadyExists.prototype);
    }
}
exports.KeySigningKeyAlreadyExists = KeySigningKeyAlreadyExists;
class TooManyKeySigningKeys extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TooManyKeySigningKeys",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyKeySigningKeys";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyKeySigningKeys.prototype);
    }
}
exports.TooManyKeySigningKeys = TooManyKeySigningKeys;
var CreateQueryLoggingConfigRequest;
(function (CreateQueryLoggingConfigRequest) {
    CreateQueryLoggingConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateQueryLoggingConfigRequest = exports.CreateQueryLoggingConfigRequest || (exports.CreateQueryLoggingConfigRequest = {}));
var QueryLoggingConfig;
(function (QueryLoggingConfig) {
    QueryLoggingConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(QueryLoggingConfig = exports.QueryLoggingConfig || (exports.QueryLoggingConfig = {}));
var CreateQueryLoggingConfigResponse;
(function (CreateQueryLoggingConfigResponse) {
    CreateQueryLoggingConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateQueryLoggingConfigResponse = exports.CreateQueryLoggingConfigResponse || (exports.CreateQueryLoggingConfigResponse = {}));
class InsufficientCloudWatchLogsResourcePolicy extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InsufficientCloudWatchLogsResourcePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "InsufficientCloudWatchLogsResourcePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, InsufficientCloudWatchLogsResourcePolicy.prototype);
    }
}
exports.InsufficientCloudWatchLogsResourcePolicy = InsufficientCloudWatchLogsResourcePolicy;
class NoSuchCloudWatchLogsLogGroup extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchCloudWatchLogsLogGroup",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchCloudWatchLogsLogGroup";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchCloudWatchLogsLogGroup.prototype);
    }
}
exports.NoSuchCloudWatchLogsLogGroup = NoSuchCloudWatchLogsLogGroup;
class QueryLoggingConfigAlreadyExists extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "QueryLoggingConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "QueryLoggingConfigAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, QueryLoggingConfigAlreadyExists.prototype);
    }
}
exports.QueryLoggingConfigAlreadyExists = QueryLoggingConfigAlreadyExists;
var CreateReusableDelegationSetRequest;
(function (CreateReusableDelegationSetRequest) {
    CreateReusableDelegationSetRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateReusableDelegationSetRequest = exports.CreateReusableDelegationSetRequest || (exports.CreateReusableDelegationSetRequest = {}));
var CreateReusableDelegationSetResponse;
(function (CreateReusableDelegationSetResponse) {
    CreateReusableDelegationSetResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateReusableDelegationSetResponse = exports.CreateReusableDelegationSetResponse || (exports.CreateReusableDelegationSetResponse = {}));
class DelegationSetAlreadyCreated extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "DelegationSetAlreadyCreated",
            $fault: "client",
            ...opts,
        });
        this.name = "DelegationSetAlreadyCreated";
        this.$fault = "client";
        Object.setPrototypeOf(this, DelegationSetAlreadyCreated.prototype);
    }
}
exports.DelegationSetAlreadyCreated = DelegationSetAlreadyCreated;
class DelegationSetAlreadyReusable extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "DelegationSetAlreadyReusable",
            $fault: "client",
            ...opts,
        });
        this.name = "DelegationSetAlreadyReusable";
        this.$fault = "client";
        Object.setPrototypeOf(this, DelegationSetAlreadyReusable.prototype);
    }
}
exports.DelegationSetAlreadyReusable = DelegationSetAlreadyReusable;
class HostedZoneNotFound extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "HostedZoneNotFound",
            $fault: "client",
            ...opts,
        });
        this.name = "HostedZoneNotFound";
        this.$fault = "client";
        Object.setPrototypeOf(this, HostedZoneNotFound.prototype);
    }
}
exports.HostedZoneNotFound = HostedZoneNotFound;
var CreateTrafficPolicyRequest;
(function (CreateTrafficPolicyRequest) {
    CreateTrafficPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateTrafficPolicyRequest = exports.CreateTrafficPolicyRequest || (exports.CreateTrafficPolicyRequest = {}));
var TrafficPolicy;
(function (TrafficPolicy) {
    TrafficPolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TrafficPolicy = exports.TrafficPolicy || (exports.TrafficPolicy = {}));
var CreateTrafficPolicyResponse;
(function (CreateTrafficPolicyResponse) {
    CreateTrafficPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateTrafficPolicyResponse = exports.CreateTrafficPolicyResponse || (exports.CreateTrafficPolicyResponse = {}));
class InvalidTrafficPolicyDocument extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidTrafficPolicyDocument",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidTrafficPolicyDocument";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidTrafficPolicyDocument.prototype);
    }
}
exports.InvalidTrafficPolicyDocument = InvalidTrafficPolicyDocument;
class TooManyTrafficPolicies extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicies",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTrafficPolicies";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTrafficPolicies.prototype);
    }
}
exports.TooManyTrafficPolicies = TooManyTrafficPolicies;
class TrafficPolicyAlreadyExists extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TrafficPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "TrafficPolicyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, TrafficPolicyAlreadyExists.prototype);
    }
}
exports.TrafficPolicyAlreadyExists = TrafficPolicyAlreadyExists;
var CreateTrafficPolicyInstanceRequest;
(function (CreateTrafficPolicyInstanceRequest) {
    CreateTrafficPolicyInstanceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateTrafficPolicyInstanceRequest = exports.CreateTrafficPolicyInstanceRequest || (exports.CreateTrafficPolicyInstanceRequest = {}));
var TrafficPolicyInstance;
(function (TrafficPolicyInstance) {
    TrafficPolicyInstance.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TrafficPolicyInstance = exports.TrafficPolicyInstance || (exports.TrafficPolicyInstance = {}));
var CreateTrafficPolicyInstanceResponse;
(function (CreateTrafficPolicyInstanceResponse) {
    CreateTrafficPolicyInstanceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateTrafficPolicyInstanceResponse = exports.CreateTrafficPolicyInstanceResponse || (exports.CreateTrafficPolicyInstanceResponse = {}));
class NoSuchTrafficPolicy extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchTrafficPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchTrafficPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchTrafficPolicy.prototype);
    }
}
exports.NoSuchTrafficPolicy = NoSuchTrafficPolicy;
class TooManyTrafficPolicyInstances extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicyInstances",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTrafficPolicyInstances";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTrafficPolicyInstances.prototype);
    }
}
exports.TooManyTrafficPolicyInstances = TooManyTrafficPolicyInstances;
class TrafficPolicyInstanceAlreadyExists extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TrafficPolicyInstanceAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "TrafficPolicyInstanceAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, TrafficPolicyInstanceAlreadyExists.prototype);
    }
}
exports.TrafficPolicyInstanceAlreadyExists = TrafficPolicyInstanceAlreadyExists;
var CreateTrafficPolicyVersionRequest;
(function (CreateTrafficPolicyVersionRequest) {
    CreateTrafficPolicyVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateTrafficPolicyVersionRequest = exports.CreateTrafficPolicyVersionRequest || (exports.CreateTrafficPolicyVersionRequest = {}));
var CreateTrafficPolicyVersionResponse;
(function (CreateTrafficPolicyVersionResponse) {
    CreateTrafficPolicyVersionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateTrafficPolicyVersionResponse = exports.CreateTrafficPolicyVersionResponse || (exports.CreateTrafficPolicyVersionResponse = {}));
class TooManyTrafficPolicyVersionsForCurrentPolicy extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicyVersionsForCurrentPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTrafficPolicyVersionsForCurrentPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTrafficPolicyVersionsForCurrentPolicy.prototype);
    }
}
exports.TooManyTrafficPolicyVersionsForCurrentPolicy = TooManyTrafficPolicyVersionsForCurrentPolicy;
var CreateVPCAssociationAuthorizationRequest;
(function (CreateVPCAssociationAuthorizationRequest) {
    CreateVPCAssociationAuthorizationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateVPCAssociationAuthorizationRequest = exports.CreateVPCAssociationAuthorizationRequest || (exports.CreateVPCAssociationAuthorizationRequest = {}));
var CreateVPCAssociationAuthorizationResponse;
(function (CreateVPCAssociationAuthorizationResponse) {
    CreateVPCAssociationAuthorizationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateVPCAssociationAuthorizationResponse = exports.CreateVPCAssociationAuthorizationResponse || (exports.CreateVPCAssociationAuthorizationResponse = {}));
class TooManyVPCAssociationAuthorizations extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TooManyVPCAssociationAuthorizations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyVPCAssociationAuthorizations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyVPCAssociationAuthorizations.prototype);
    }
}
exports.TooManyVPCAssociationAuthorizations = TooManyVPCAssociationAuthorizations;
var DeactivateKeySigningKeyRequest;
(function (DeactivateKeySigningKeyRequest) {
    DeactivateKeySigningKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeactivateKeySigningKeyRequest = exports.DeactivateKeySigningKeyRequest || (exports.DeactivateKeySigningKeyRequest = {}));
var DeactivateKeySigningKeyResponse;
(function (DeactivateKeySigningKeyResponse) {
    DeactivateKeySigningKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeactivateKeySigningKeyResponse = exports.DeactivateKeySigningKeyResponse || (exports.DeactivateKeySigningKeyResponse = {}));
class KeySigningKeyInParentDSRecord extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "KeySigningKeyInParentDSRecord",
            $fault: "client",
            ...opts,
        });
        this.name = "KeySigningKeyInParentDSRecord";
        this.$fault = "client";
        Object.setPrototypeOf(this, KeySigningKeyInParentDSRecord.prototype);
    }
}
exports.KeySigningKeyInParentDSRecord = KeySigningKeyInParentDSRecord;
class KeySigningKeyInUse extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "KeySigningKeyInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "KeySigningKeyInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, KeySigningKeyInUse.prototype);
    }
}
exports.KeySigningKeyInUse = KeySigningKeyInUse;
class CidrCollectionInUseException extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "CidrCollectionInUseException",
            $fault: "client",
            ...opts,
        });
        this.name = "CidrCollectionInUseException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CidrCollectionInUseException.prototype);
        this.Message = opts.Message;
    }
}
exports.CidrCollectionInUseException = CidrCollectionInUseException;
var DeleteCidrCollectionRequest;
(function (DeleteCidrCollectionRequest) {
    DeleteCidrCollectionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCidrCollectionRequest = exports.DeleteCidrCollectionRequest || (exports.DeleteCidrCollectionRequest = {}));
var DeleteCidrCollectionResponse;
(function (DeleteCidrCollectionResponse) {
    DeleteCidrCollectionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCidrCollectionResponse = exports.DeleteCidrCollectionResponse || (exports.DeleteCidrCollectionResponse = {}));
var DeleteHealthCheckRequest;
(function (DeleteHealthCheckRequest) {
    DeleteHealthCheckRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteHealthCheckRequest = exports.DeleteHealthCheckRequest || (exports.DeleteHealthCheckRequest = {}));
var DeleteHealthCheckResponse;
(function (DeleteHealthCheckResponse) {
    DeleteHealthCheckResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteHealthCheckResponse = exports.DeleteHealthCheckResponse || (exports.DeleteHealthCheckResponse = {}));
class HealthCheckInUse extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "HealthCheckInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "HealthCheckInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, HealthCheckInUse.prototype);
    }
}
exports.HealthCheckInUse = HealthCheckInUse;
var DeleteHostedZoneRequest;
(function (DeleteHostedZoneRequest) {
    DeleteHostedZoneRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteHostedZoneRequest = exports.DeleteHostedZoneRequest || (exports.DeleteHostedZoneRequest = {}));
var DeleteHostedZoneResponse;
(function (DeleteHostedZoneResponse) {
    DeleteHostedZoneResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteHostedZoneResponse = exports.DeleteHostedZoneResponse || (exports.DeleteHostedZoneResponse = {}));
class HostedZoneNotEmpty extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "HostedZoneNotEmpty",
            $fault: "client",
            ...opts,
        });
        this.name = "HostedZoneNotEmpty";
        this.$fault = "client";
        Object.setPrototypeOf(this, HostedZoneNotEmpty.prototype);
    }
}
exports.HostedZoneNotEmpty = HostedZoneNotEmpty;
var DeleteKeySigningKeyRequest;
(function (DeleteKeySigningKeyRequest) {
    DeleteKeySigningKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteKeySigningKeyRequest = exports.DeleteKeySigningKeyRequest || (exports.DeleteKeySigningKeyRequest = {}));
var DeleteKeySigningKeyResponse;
(function (DeleteKeySigningKeyResponse) {
    DeleteKeySigningKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteKeySigningKeyResponse = exports.DeleteKeySigningKeyResponse || (exports.DeleteKeySigningKeyResponse = {}));
var DeleteQueryLoggingConfigRequest;
(function (DeleteQueryLoggingConfigRequest) {
    DeleteQueryLoggingConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteQueryLoggingConfigRequest = exports.DeleteQueryLoggingConfigRequest || (exports.DeleteQueryLoggingConfigRequest = {}));
var DeleteQueryLoggingConfigResponse;
(function (DeleteQueryLoggingConfigResponse) {
    DeleteQueryLoggingConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteQueryLoggingConfigResponse = exports.DeleteQueryLoggingConfigResponse || (exports.DeleteQueryLoggingConfigResponse = {}));
class NoSuchQueryLoggingConfig extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchQueryLoggingConfig",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchQueryLoggingConfig";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchQueryLoggingConfig.prototype);
    }
}
exports.NoSuchQueryLoggingConfig = NoSuchQueryLoggingConfig;
class DelegationSetInUse extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "DelegationSetInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "DelegationSetInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, DelegationSetInUse.prototype);
    }
}
exports.DelegationSetInUse = DelegationSetInUse;
var DeleteReusableDelegationSetRequest;
(function (DeleteReusableDelegationSetRequest) {
    DeleteReusableDelegationSetRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteReusableDelegationSetRequest = exports.DeleteReusableDelegationSetRequest || (exports.DeleteReusableDelegationSetRequest = {}));
var DeleteReusableDelegationSetResponse;
(function (DeleteReusableDelegationSetResponse) {
    DeleteReusableDelegationSetResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteReusableDelegationSetResponse = exports.DeleteReusableDelegationSetResponse || (exports.DeleteReusableDelegationSetResponse = {}));
var DeleteTrafficPolicyRequest;
(function (DeleteTrafficPolicyRequest) {
    DeleteTrafficPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteTrafficPolicyRequest = exports.DeleteTrafficPolicyRequest || (exports.DeleteTrafficPolicyRequest = {}));
var DeleteTrafficPolicyResponse;
(function (DeleteTrafficPolicyResponse) {
    DeleteTrafficPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteTrafficPolicyResponse = exports.DeleteTrafficPolicyResponse || (exports.DeleteTrafficPolicyResponse = {}));
class TrafficPolicyInUse extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "TrafficPolicyInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "TrafficPolicyInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, TrafficPolicyInUse.prototype);
    }
}
exports.TrafficPolicyInUse = TrafficPolicyInUse;
var DeleteTrafficPolicyInstanceRequest;
(function (DeleteTrafficPolicyInstanceRequest) {
    DeleteTrafficPolicyInstanceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteTrafficPolicyInstanceRequest = exports.DeleteTrafficPolicyInstanceRequest || (exports.DeleteTrafficPolicyInstanceRequest = {}));
var DeleteTrafficPolicyInstanceResponse;
(function (DeleteTrafficPolicyInstanceResponse) {
    DeleteTrafficPolicyInstanceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteTrafficPolicyInstanceResponse = exports.DeleteTrafficPolicyInstanceResponse || (exports.DeleteTrafficPolicyInstanceResponse = {}));
class NoSuchTrafficPolicyInstance extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchTrafficPolicyInstance",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchTrafficPolicyInstance";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchTrafficPolicyInstance.prototype);
    }
}
exports.NoSuchTrafficPolicyInstance = NoSuchTrafficPolicyInstance;
var DeleteVPCAssociationAuthorizationRequest;
(function (DeleteVPCAssociationAuthorizationRequest) {
    DeleteVPCAssociationAuthorizationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteVPCAssociationAuthorizationRequest = exports.DeleteVPCAssociationAuthorizationRequest || (exports.DeleteVPCAssociationAuthorizationRequest = {}));
var DeleteVPCAssociationAuthorizationResponse;
(function (DeleteVPCAssociationAuthorizationResponse) {
    DeleteVPCAssociationAuthorizationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteVPCAssociationAuthorizationResponse = exports.DeleteVPCAssociationAuthorizationResponse || (exports.DeleteVPCAssociationAuthorizationResponse = {}));
class VPCAssociationAuthorizationNotFound extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "VPCAssociationAuthorizationNotFound",
            $fault: "client",
            ...opts,
        });
        this.name = "VPCAssociationAuthorizationNotFound";
        this.$fault = "client";
        Object.setPrototypeOf(this, VPCAssociationAuthorizationNotFound.prototype);
    }
}
exports.VPCAssociationAuthorizationNotFound = VPCAssociationAuthorizationNotFound;
var DisableHostedZoneDNSSECRequest;
(function (DisableHostedZoneDNSSECRequest) {
    DisableHostedZoneDNSSECRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisableHostedZoneDNSSECRequest = exports.DisableHostedZoneDNSSECRequest || (exports.DisableHostedZoneDNSSECRequest = {}));
var DisableHostedZoneDNSSECResponse;
(function (DisableHostedZoneDNSSECResponse) {
    DisableHostedZoneDNSSECResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisableHostedZoneDNSSECResponse = exports.DisableHostedZoneDNSSECResponse || (exports.DisableHostedZoneDNSSECResponse = {}));
class DNSSECNotFound extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "DNSSECNotFound",
            $fault: "client",
            ...opts,
        });
        this.name = "DNSSECNotFound";
        this.$fault = "client";
        Object.setPrototypeOf(this, DNSSECNotFound.prototype);
    }
}
exports.DNSSECNotFound = DNSSECNotFound;
var DisassociateVPCFromHostedZoneRequest;
(function (DisassociateVPCFromHostedZoneRequest) {
    DisassociateVPCFromHostedZoneRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisassociateVPCFromHostedZoneRequest = exports.DisassociateVPCFromHostedZoneRequest || (exports.DisassociateVPCFromHostedZoneRequest = {}));
var DisassociateVPCFromHostedZoneResponse;
(function (DisassociateVPCFromHostedZoneResponse) {
    DisassociateVPCFromHostedZoneResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisassociateVPCFromHostedZoneResponse = exports.DisassociateVPCFromHostedZoneResponse || (exports.DisassociateVPCFromHostedZoneResponse = {}));
class LastVPCAssociation extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "LastVPCAssociation",
            $fault: "client",
            ...opts,
        });
        this.name = "LastVPCAssociation";
        this.$fault = "client";
        Object.setPrototypeOf(this, LastVPCAssociation.prototype);
    }
}
exports.LastVPCAssociation = LastVPCAssociation;
class VPCAssociationNotFound extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "VPCAssociationNotFound",
            $fault: "client",
            ...opts,
        });
        this.name = "VPCAssociationNotFound";
        this.$fault = "client";
        Object.setPrototypeOf(this, VPCAssociationNotFound.prototype);
    }
}
exports.VPCAssociationNotFound = VPCAssociationNotFound;
var EnableHostedZoneDNSSECRequest;
(function (EnableHostedZoneDNSSECRequest) {
    EnableHostedZoneDNSSECRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EnableHostedZoneDNSSECRequest = exports.EnableHostedZoneDNSSECRequest || (exports.EnableHostedZoneDNSSECRequest = {}));
var EnableHostedZoneDNSSECResponse;
(function (EnableHostedZoneDNSSECResponse) {
    EnableHostedZoneDNSSECResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EnableHostedZoneDNSSECResponse = exports.EnableHostedZoneDNSSECResponse || (exports.EnableHostedZoneDNSSECResponse = {}));
class HostedZonePartiallyDelegated extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "HostedZonePartiallyDelegated",
            $fault: "client",
            ...opts,
        });
        this.name = "HostedZonePartiallyDelegated";
        this.$fault = "client";
        Object.setPrototypeOf(this, HostedZonePartiallyDelegated.prototype);
    }
}
exports.HostedZonePartiallyDelegated = HostedZonePartiallyDelegated;
class KeySigningKeyWithActiveStatusNotFound extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "KeySigningKeyWithActiveStatusNotFound",
            $fault: "client",
            ...opts,
        });
        this.name = "KeySigningKeyWithActiveStatusNotFound";
        this.$fault = "client";
        Object.setPrototypeOf(this, KeySigningKeyWithActiveStatusNotFound.prototype);
    }
}
exports.KeySigningKeyWithActiveStatusNotFound = KeySigningKeyWithActiveStatusNotFound;
var GetAccountLimitRequest;
(function (GetAccountLimitRequest) {
    GetAccountLimitRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccountLimitRequest = exports.GetAccountLimitRequest || (exports.GetAccountLimitRequest = {}));
var GetAccountLimitResponse;
(function (GetAccountLimitResponse) {
    GetAccountLimitResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetAccountLimitResponse = exports.GetAccountLimitResponse || (exports.GetAccountLimitResponse = {}));
var GetChangeRequest;
(function (GetChangeRequest) {
    GetChangeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetChangeRequest = exports.GetChangeRequest || (exports.GetChangeRequest = {}));
var GetChangeResponse;
(function (GetChangeResponse) {
    GetChangeResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetChangeResponse = exports.GetChangeResponse || (exports.GetChangeResponse = {}));
class NoSuchChange extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchChange",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchChange";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchChange.prototype);
    }
}
exports.NoSuchChange = NoSuchChange;
var GetCheckerIpRangesRequest;
(function (GetCheckerIpRangesRequest) {
    GetCheckerIpRangesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCheckerIpRangesRequest = exports.GetCheckerIpRangesRequest || (exports.GetCheckerIpRangesRequest = {}));
var GetCheckerIpRangesResponse;
(function (GetCheckerIpRangesResponse) {
    GetCheckerIpRangesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCheckerIpRangesResponse = exports.GetCheckerIpRangesResponse || (exports.GetCheckerIpRangesResponse = {}));
var GetDNSSECRequest;
(function (GetDNSSECRequest) {
    GetDNSSECRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetDNSSECRequest = exports.GetDNSSECRequest || (exports.GetDNSSECRequest = {}));
var DNSSECStatus;
(function (DNSSECStatus) {
    DNSSECStatus.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DNSSECStatus = exports.DNSSECStatus || (exports.DNSSECStatus = {}));
var GetDNSSECResponse;
(function (GetDNSSECResponse) {
    GetDNSSECResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetDNSSECResponse = exports.GetDNSSECResponse || (exports.GetDNSSECResponse = {}));
var GetGeoLocationRequest;
(function (GetGeoLocationRequest) {
    GetGeoLocationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetGeoLocationRequest = exports.GetGeoLocationRequest || (exports.GetGeoLocationRequest = {}));
var GeoLocationDetails;
(function (GeoLocationDetails) {
    GeoLocationDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GeoLocationDetails = exports.GeoLocationDetails || (exports.GeoLocationDetails = {}));
var GetGeoLocationResponse;
(function (GetGeoLocationResponse) {
    GetGeoLocationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetGeoLocationResponse = exports.GetGeoLocationResponse || (exports.GetGeoLocationResponse = {}));
class NoSuchGeoLocation extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchGeoLocation",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchGeoLocation";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchGeoLocation.prototype);
    }
}
exports.NoSuchGeoLocation = NoSuchGeoLocation;
var GetHealthCheckRequest;
(function (GetHealthCheckRequest) {
    GetHealthCheckRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHealthCheckRequest = exports.GetHealthCheckRequest || (exports.GetHealthCheckRequest = {}));
var GetHealthCheckResponse;
(function (GetHealthCheckResponse) {
    GetHealthCheckResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHealthCheckResponse = exports.GetHealthCheckResponse || (exports.GetHealthCheckResponse = {}));
class IncompatibleVersion extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "IncompatibleVersion",
            $fault: "client",
            ...opts,
        });
        this.name = "IncompatibleVersion";
        this.$fault = "client";
        Object.setPrototypeOf(this, IncompatibleVersion.prototype);
    }
}
exports.IncompatibleVersion = IncompatibleVersion;
var GetHealthCheckCountRequest;
(function (GetHealthCheckCountRequest) {
    GetHealthCheckCountRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHealthCheckCountRequest = exports.GetHealthCheckCountRequest || (exports.GetHealthCheckCountRequest = {}));
var GetHealthCheckCountResponse;
(function (GetHealthCheckCountResponse) {
    GetHealthCheckCountResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHealthCheckCountResponse = exports.GetHealthCheckCountResponse || (exports.GetHealthCheckCountResponse = {}));
var GetHealthCheckLastFailureReasonRequest;
(function (GetHealthCheckLastFailureReasonRequest) {
    GetHealthCheckLastFailureReasonRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHealthCheckLastFailureReasonRequest = exports.GetHealthCheckLastFailureReasonRequest || (exports.GetHealthCheckLastFailureReasonRequest = {}));
var StatusReport;
(function (StatusReport) {
    StatusReport.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StatusReport = exports.StatusReport || (exports.StatusReport = {}));
var HealthCheckObservation;
(function (HealthCheckObservation) {
    HealthCheckObservation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HealthCheckObservation = exports.HealthCheckObservation || (exports.HealthCheckObservation = {}));
var GetHealthCheckLastFailureReasonResponse;
(function (GetHealthCheckLastFailureReasonResponse) {
    GetHealthCheckLastFailureReasonResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHealthCheckLastFailureReasonResponse = exports.GetHealthCheckLastFailureReasonResponse || (exports.GetHealthCheckLastFailureReasonResponse = {}));
var GetHealthCheckStatusRequest;
(function (GetHealthCheckStatusRequest) {
    GetHealthCheckStatusRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHealthCheckStatusRequest = exports.GetHealthCheckStatusRequest || (exports.GetHealthCheckStatusRequest = {}));
var GetHealthCheckStatusResponse;
(function (GetHealthCheckStatusResponse) {
    GetHealthCheckStatusResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHealthCheckStatusResponse = exports.GetHealthCheckStatusResponse || (exports.GetHealthCheckStatusResponse = {}));
var GetHostedZoneRequest;
(function (GetHostedZoneRequest) {
    GetHostedZoneRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHostedZoneRequest = exports.GetHostedZoneRequest || (exports.GetHostedZoneRequest = {}));
var GetHostedZoneResponse;
(function (GetHostedZoneResponse) {
    GetHostedZoneResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHostedZoneResponse = exports.GetHostedZoneResponse || (exports.GetHostedZoneResponse = {}));
var GetHostedZoneCountRequest;
(function (GetHostedZoneCountRequest) {
    GetHostedZoneCountRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHostedZoneCountRequest = exports.GetHostedZoneCountRequest || (exports.GetHostedZoneCountRequest = {}));
var GetHostedZoneCountResponse;
(function (GetHostedZoneCountResponse) {
    GetHostedZoneCountResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHostedZoneCountResponse = exports.GetHostedZoneCountResponse || (exports.GetHostedZoneCountResponse = {}));
var GetHostedZoneLimitRequest;
(function (GetHostedZoneLimitRequest) {
    GetHostedZoneLimitRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHostedZoneLimitRequest = exports.GetHostedZoneLimitRequest || (exports.GetHostedZoneLimitRequest = {}));
var HostedZoneLimit;
(function (HostedZoneLimit) {
    HostedZoneLimit.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HostedZoneLimit = exports.HostedZoneLimit || (exports.HostedZoneLimit = {}));
var GetHostedZoneLimitResponse;
(function (GetHostedZoneLimitResponse) {
    GetHostedZoneLimitResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetHostedZoneLimitResponse = exports.GetHostedZoneLimitResponse || (exports.GetHostedZoneLimitResponse = {}));
class HostedZoneNotPrivate extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "HostedZoneNotPrivate",
            $fault: "client",
            ...opts,
        });
        this.name = "HostedZoneNotPrivate";
        this.$fault = "client";
        Object.setPrototypeOf(this, HostedZoneNotPrivate.prototype);
    }
}
exports.HostedZoneNotPrivate = HostedZoneNotPrivate;
var GetQueryLoggingConfigRequest;
(function (GetQueryLoggingConfigRequest) {
    GetQueryLoggingConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetQueryLoggingConfigRequest = exports.GetQueryLoggingConfigRequest || (exports.GetQueryLoggingConfigRequest = {}));
var GetQueryLoggingConfigResponse;
(function (GetQueryLoggingConfigResponse) {
    GetQueryLoggingConfigResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetQueryLoggingConfigResponse = exports.GetQueryLoggingConfigResponse || (exports.GetQueryLoggingConfigResponse = {}));
var GetReusableDelegationSetRequest;
(function (GetReusableDelegationSetRequest) {
    GetReusableDelegationSetRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetReusableDelegationSetRequest = exports.GetReusableDelegationSetRequest || (exports.GetReusableDelegationSetRequest = {}));
var GetReusableDelegationSetResponse;
(function (GetReusableDelegationSetResponse) {
    GetReusableDelegationSetResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetReusableDelegationSetResponse = exports.GetReusableDelegationSetResponse || (exports.GetReusableDelegationSetResponse = {}));
var GetReusableDelegationSetLimitRequest;
(function (GetReusableDelegationSetLimitRequest) {
    GetReusableDelegationSetLimitRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetReusableDelegationSetLimitRequest = exports.GetReusableDelegationSetLimitRequest || (exports.GetReusableDelegationSetLimitRequest = {}));
var ReusableDelegationSetLimit;
(function (ReusableDelegationSetLimit) {
    ReusableDelegationSetLimit.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ReusableDelegationSetLimit = exports.ReusableDelegationSetLimit || (exports.ReusableDelegationSetLimit = {}));
var GetReusableDelegationSetLimitResponse;
(function (GetReusableDelegationSetLimitResponse) {
    GetReusableDelegationSetLimitResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetReusableDelegationSetLimitResponse = exports.GetReusableDelegationSetLimitResponse || (exports.GetReusableDelegationSetLimitResponse = {}));
var GetTrafficPolicyRequest;
(function (GetTrafficPolicyRequest) {
    GetTrafficPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTrafficPolicyRequest = exports.GetTrafficPolicyRequest || (exports.GetTrafficPolicyRequest = {}));
var GetTrafficPolicyResponse;
(function (GetTrafficPolicyResponse) {
    GetTrafficPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTrafficPolicyResponse = exports.GetTrafficPolicyResponse || (exports.GetTrafficPolicyResponse = {}));
var GetTrafficPolicyInstanceRequest;
(function (GetTrafficPolicyInstanceRequest) {
    GetTrafficPolicyInstanceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTrafficPolicyInstanceRequest = exports.GetTrafficPolicyInstanceRequest || (exports.GetTrafficPolicyInstanceRequest = {}));
var GetTrafficPolicyInstanceResponse;
(function (GetTrafficPolicyInstanceResponse) {
    GetTrafficPolicyInstanceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTrafficPolicyInstanceResponse = exports.GetTrafficPolicyInstanceResponse || (exports.GetTrafficPolicyInstanceResponse = {}));
var GetTrafficPolicyInstanceCountRequest;
(function (GetTrafficPolicyInstanceCountRequest) {
    GetTrafficPolicyInstanceCountRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTrafficPolicyInstanceCountRequest = exports.GetTrafficPolicyInstanceCountRequest || (exports.GetTrafficPolicyInstanceCountRequest = {}));
var GetTrafficPolicyInstanceCountResponse;
(function (GetTrafficPolicyInstanceCountResponse) {
    GetTrafficPolicyInstanceCountResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTrafficPolicyInstanceCountResponse = exports.GetTrafficPolicyInstanceCountResponse || (exports.GetTrafficPolicyInstanceCountResponse = {}));
var ListCidrBlocksRequest;
(function (ListCidrBlocksRequest) {
    ListCidrBlocksRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCidrBlocksRequest = exports.ListCidrBlocksRequest || (exports.ListCidrBlocksRequest = {}));
var CidrBlockSummary;
(function (CidrBlockSummary) {
    CidrBlockSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CidrBlockSummary = exports.CidrBlockSummary || (exports.CidrBlockSummary = {}));
var ListCidrBlocksResponse;
(function (ListCidrBlocksResponse) {
    ListCidrBlocksResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCidrBlocksResponse = exports.ListCidrBlocksResponse || (exports.ListCidrBlocksResponse = {}));
class NoSuchCidrLocationException extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "NoSuchCidrLocationException",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchCidrLocationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchCidrLocationException.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchCidrLocationException = NoSuchCidrLocationException;
var ListCidrCollectionsRequest;
(function (ListCidrCollectionsRequest) {
    ListCidrCollectionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCidrCollectionsRequest = exports.ListCidrCollectionsRequest || (exports.ListCidrCollectionsRequest = {}));
var CollectionSummary;
(function (CollectionSummary) {
    CollectionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CollectionSummary = exports.CollectionSummary || (exports.CollectionSummary = {}));
var ListCidrCollectionsResponse;
(function (ListCidrCollectionsResponse) {
    ListCidrCollectionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCidrCollectionsResponse = exports.ListCidrCollectionsResponse || (exports.ListCidrCollectionsResponse = {}));
var ListCidrLocationsRequest;
(function (ListCidrLocationsRequest) {
    ListCidrLocationsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCidrLocationsRequest = exports.ListCidrLocationsRequest || (exports.ListCidrLocationsRequest = {}));
var LocationSummary;
(function (LocationSummary) {
    LocationSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LocationSummary = exports.LocationSummary || (exports.LocationSummary = {}));
var ListCidrLocationsResponse;
(function (ListCidrLocationsResponse) {
    ListCidrLocationsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCidrLocationsResponse = exports.ListCidrLocationsResponse || (exports.ListCidrLocationsResponse = {}));
var ListGeoLocationsRequest;
(function (ListGeoLocationsRequest) {
    ListGeoLocationsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGeoLocationsRequest = exports.ListGeoLocationsRequest || (exports.ListGeoLocationsRequest = {}));
var ListGeoLocationsResponse;
(function (ListGeoLocationsResponse) {
    ListGeoLocationsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGeoLocationsResponse = exports.ListGeoLocationsResponse || (exports.ListGeoLocationsResponse = {}));
var ListHealthChecksRequest;
(function (ListHealthChecksRequest) {
    ListHealthChecksRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListHealthChecksRequest = exports.ListHealthChecksRequest || (exports.ListHealthChecksRequest = {}));
var ListHealthChecksResponse;
(function (ListHealthChecksResponse) {
    ListHealthChecksResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListHealthChecksResponse = exports.ListHealthChecksResponse || (exports.ListHealthChecksResponse = {}));
var ListHostedZonesRequest;
(function (ListHostedZonesRequest) {
    ListHostedZonesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListHostedZonesRequest = exports.ListHostedZonesRequest || (exports.ListHostedZonesRequest = {}));
var ListHostedZonesResponse;
(function (ListHostedZonesResponse) {
    ListHostedZonesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListHostedZonesResponse = exports.ListHostedZonesResponse || (exports.ListHostedZonesResponse = {}));
var ListHostedZonesByNameRequest;
(function (ListHostedZonesByNameRequest) {
    ListHostedZonesByNameRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListHostedZonesByNameRequest = exports.ListHostedZonesByNameRequest || (exports.ListHostedZonesByNameRequest = {}));
var ListHostedZonesByNameResponse;
(function (ListHostedZonesByNameResponse) {
    ListHostedZonesByNameResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListHostedZonesByNameResponse = exports.ListHostedZonesByNameResponse || (exports.ListHostedZonesByNameResponse = {}));
class InvalidPaginationToken extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "InvalidPaginationToken",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidPaginationToken";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidPaginationToken.prototype);
    }
}
exports.InvalidPaginationToken = InvalidPaginationToken;
var ListHostedZonesByVPCRequest;
(function (ListHostedZonesByVPCRequest) {
    ListHostedZonesByVPCRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListHostedZonesByVPCRequest = exports.ListHostedZonesByVPCRequest || (exports.ListHostedZonesByVPCRequest = {}));
var HostedZoneOwner;
(function (HostedZoneOwner) {
    HostedZoneOwner.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HostedZoneOwner = exports.HostedZoneOwner || (exports.HostedZoneOwner = {}));
var HostedZoneSummary;
(function (HostedZoneSummary) {
    HostedZoneSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(HostedZoneSummary = exports.HostedZoneSummary || (exports.HostedZoneSummary = {}));
var ListHostedZonesByVPCResponse;
(function (ListHostedZonesByVPCResponse) {
    ListHostedZonesByVPCResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListHostedZonesByVPCResponse = exports.ListHostedZonesByVPCResponse || (exports.ListHostedZonesByVPCResponse = {}));
var ListQueryLoggingConfigsRequest;
(function (ListQueryLoggingConfigsRequest) {
    ListQueryLoggingConfigsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListQueryLoggingConfigsRequest = exports.ListQueryLoggingConfigsRequest || (exports.ListQueryLoggingConfigsRequest = {}));
var ListQueryLoggingConfigsResponse;
(function (ListQueryLoggingConfigsResponse) {
    ListQueryLoggingConfigsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListQueryLoggingConfigsResponse = exports.ListQueryLoggingConfigsResponse || (exports.ListQueryLoggingConfigsResponse = {}));
var ListResourceRecordSetsRequest;
(function (ListResourceRecordSetsRequest) {
    ListResourceRecordSetsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListResourceRecordSetsRequest = exports.ListResourceRecordSetsRequest || (exports.ListResourceRecordSetsRequest = {}));
var ListResourceRecordSetsResponse;
(function (ListResourceRecordSetsResponse) {
    ListResourceRecordSetsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListResourceRecordSetsResponse = exports.ListResourceRecordSetsResponse || (exports.ListResourceRecordSetsResponse = {}));
var ListReusableDelegationSetsRequest;
(function (ListReusableDelegationSetsRequest) {
    ListReusableDelegationSetsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListReusableDelegationSetsRequest = exports.ListReusableDelegationSetsRequest || (exports.ListReusableDelegationSetsRequest = {}));
var ListReusableDelegationSetsResponse;
(function (ListReusableDelegationSetsResponse) {
    ListReusableDelegationSetsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListReusableDelegationSetsResponse = exports.ListReusableDelegationSetsResponse || (exports.ListReusableDelegationSetsResponse = {}));
var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceRequest = exports.ListTagsForResourceRequest || (exports.ListTagsForResourceRequest = {}));
var ResourceTagSet;
(function (ResourceTagSet) {
    ResourceTagSet.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceTagSet = exports.ResourceTagSet || (exports.ResourceTagSet = {}));
var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceResponse = exports.ListTagsForResourceResponse || (exports.ListTagsForResourceResponse = {}));
var ListTagsForResourcesRequest;
(function (ListTagsForResourcesRequest) {
    ListTagsForResourcesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourcesRequest = exports.ListTagsForResourcesRequest || (exports.ListTagsForResourcesRequest = {}));
var ListTagsForResourcesResponse;
(function (ListTagsForResourcesResponse) {
    ListTagsForResourcesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourcesResponse = exports.ListTagsForResourcesResponse || (exports.ListTagsForResourcesResponse = {}));
var ListTrafficPoliciesRequest;
(function (ListTrafficPoliciesRequest) {
    ListTrafficPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPoliciesRequest = exports.ListTrafficPoliciesRequest || (exports.ListTrafficPoliciesRequest = {}));
var TrafficPolicySummary;
(function (TrafficPolicySummary) {
    TrafficPolicySummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TrafficPolicySummary = exports.TrafficPolicySummary || (exports.TrafficPolicySummary = {}));
var ListTrafficPoliciesResponse;
(function (ListTrafficPoliciesResponse) {
    ListTrafficPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPoliciesResponse = exports.ListTrafficPoliciesResponse || (exports.ListTrafficPoliciesResponse = {}));
var ListTrafficPolicyInstancesRequest;
(function (ListTrafficPolicyInstancesRequest) {
    ListTrafficPolicyInstancesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPolicyInstancesRequest = exports.ListTrafficPolicyInstancesRequest || (exports.ListTrafficPolicyInstancesRequest = {}));
var ListTrafficPolicyInstancesResponse;
(function (ListTrafficPolicyInstancesResponse) {
    ListTrafficPolicyInstancesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPolicyInstancesResponse = exports.ListTrafficPolicyInstancesResponse || (exports.ListTrafficPolicyInstancesResponse = {}));
var ListTrafficPolicyInstancesByHostedZoneRequest;
(function (ListTrafficPolicyInstancesByHostedZoneRequest) {
    ListTrafficPolicyInstancesByHostedZoneRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPolicyInstancesByHostedZoneRequest = exports.ListTrafficPolicyInstancesByHostedZoneRequest || (exports.ListTrafficPolicyInstancesByHostedZoneRequest = {}));
var ListTrafficPolicyInstancesByHostedZoneResponse;
(function (ListTrafficPolicyInstancesByHostedZoneResponse) {
    ListTrafficPolicyInstancesByHostedZoneResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPolicyInstancesByHostedZoneResponse = exports.ListTrafficPolicyInstancesByHostedZoneResponse || (exports.ListTrafficPolicyInstancesByHostedZoneResponse = {}));
var ListTrafficPolicyInstancesByPolicyRequest;
(function (ListTrafficPolicyInstancesByPolicyRequest) {
    ListTrafficPolicyInstancesByPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPolicyInstancesByPolicyRequest = exports.ListTrafficPolicyInstancesByPolicyRequest || (exports.ListTrafficPolicyInstancesByPolicyRequest = {}));
var ListTrafficPolicyInstancesByPolicyResponse;
(function (ListTrafficPolicyInstancesByPolicyResponse) {
    ListTrafficPolicyInstancesByPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPolicyInstancesByPolicyResponse = exports.ListTrafficPolicyInstancesByPolicyResponse || (exports.ListTrafficPolicyInstancesByPolicyResponse = {}));
var ListTrafficPolicyVersionsRequest;
(function (ListTrafficPolicyVersionsRequest) {
    ListTrafficPolicyVersionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPolicyVersionsRequest = exports.ListTrafficPolicyVersionsRequest || (exports.ListTrafficPolicyVersionsRequest = {}));
var ListTrafficPolicyVersionsResponse;
(function (ListTrafficPolicyVersionsResponse) {
    ListTrafficPolicyVersionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTrafficPolicyVersionsResponse = exports.ListTrafficPolicyVersionsResponse || (exports.ListTrafficPolicyVersionsResponse = {}));
var ListVPCAssociationAuthorizationsRequest;
(function (ListVPCAssociationAuthorizationsRequest) {
    ListVPCAssociationAuthorizationsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListVPCAssociationAuthorizationsRequest = exports.ListVPCAssociationAuthorizationsRequest || (exports.ListVPCAssociationAuthorizationsRequest = {}));
var ListVPCAssociationAuthorizationsResponse;
(function (ListVPCAssociationAuthorizationsResponse) {
    ListVPCAssociationAuthorizationsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListVPCAssociationAuthorizationsResponse = exports.ListVPCAssociationAuthorizationsResponse || (exports.ListVPCAssociationAuthorizationsResponse = {}));
var TestDNSAnswerRequest;
(function (TestDNSAnswerRequest) {
    TestDNSAnswerRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TestDNSAnswerRequest = exports.TestDNSAnswerRequest || (exports.TestDNSAnswerRequest = {}));
var TestDNSAnswerResponse;
(function (TestDNSAnswerResponse) {
    TestDNSAnswerResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TestDNSAnswerResponse = exports.TestDNSAnswerResponse || (exports.TestDNSAnswerResponse = {}));
class HealthCheckVersionMismatch extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "HealthCheckVersionMismatch",
            $fault: "client",
            ...opts,
        });
        this.name = "HealthCheckVersionMismatch";
        this.$fault = "client";
        Object.setPrototypeOf(this, HealthCheckVersionMismatch.prototype);
    }
}
exports.HealthCheckVersionMismatch = HealthCheckVersionMismatch;
var UpdateHealthCheckRequest;
(function (UpdateHealthCheckRequest) {
    UpdateHealthCheckRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateHealthCheckRequest = exports.UpdateHealthCheckRequest || (exports.UpdateHealthCheckRequest = {}));
var UpdateHealthCheckResponse;
(function (UpdateHealthCheckResponse) {
    UpdateHealthCheckResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateHealthCheckResponse = exports.UpdateHealthCheckResponse || (exports.UpdateHealthCheckResponse = {}));
var UpdateHostedZoneCommentRequest;
(function (UpdateHostedZoneCommentRequest) {
    UpdateHostedZoneCommentRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateHostedZoneCommentRequest = exports.UpdateHostedZoneCommentRequest || (exports.UpdateHostedZoneCommentRequest = {}));
var UpdateHostedZoneCommentResponse;
(function (UpdateHostedZoneCommentResponse) {
    UpdateHostedZoneCommentResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateHostedZoneCommentResponse = exports.UpdateHostedZoneCommentResponse || (exports.UpdateHostedZoneCommentResponse = {}));
var UpdateTrafficPolicyCommentRequest;
(function (UpdateTrafficPolicyCommentRequest) {
    UpdateTrafficPolicyCommentRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTrafficPolicyCommentRequest = exports.UpdateTrafficPolicyCommentRequest || (exports.UpdateTrafficPolicyCommentRequest = {}));
var UpdateTrafficPolicyCommentResponse;
(function (UpdateTrafficPolicyCommentResponse) {
    UpdateTrafficPolicyCommentResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTrafficPolicyCommentResponse = exports.UpdateTrafficPolicyCommentResponse || (exports.UpdateTrafficPolicyCommentResponse = {}));
class ConflictingTypes extends Route53ServiceException_1.Route53ServiceException {
    constructor(opts) {
        super({
            name: "ConflictingTypes",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictingTypes";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictingTypes.prototype);
    }
}
exports.ConflictingTypes = ConflictingTypes;
var UpdateTrafficPolicyInstanceRequest;
(function (UpdateTrafficPolicyInstanceRequest) {
    UpdateTrafficPolicyInstanceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTrafficPolicyInstanceRequest = exports.UpdateTrafficPolicyInstanceRequest || (exports.UpdateTrafficPolicyInstanceRequest = {}));
var UpdateTrafficPolicyInstanceResponse;
(function (UpdateTrafficPolicyInstanceResponse) {
    UpdateTrafficPolicyInstanceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTrafficPolicyInstanceResponse = exports.UpdateTrafficPolicyInstanceResponse || (exports.UpdateTrafficPolicyInstanceResponse = {}));
